From a8b6fc9c5d88df464c880380ac36a9f079e086b0 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Tue, 6 Sep 2016 00:06:04 +0200
Subject: [PATCH 0475/1345] fix: arm64: dts: cp110: fix clock dependencies for
 slow interfaces

commit  0951b5c25587abd9caf40ae45ef0d3c62b46132d from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

I2C and SPI interfaces share common clock trees within their
CP110 block. This patch fixes those dependencies for CP110-master
and slave.

Change-Id: I102ec8e48e5c52e1f91f5a4aa7c18c67446e21ca
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32578
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi |    8 ++++----
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   |    2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 7570485..b90b528 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -81,7 +81,7 @@ spi@700600 {
 	#address-cells = <0x1>;
 	#size-cells = <0x0>;
 	cell-index = <0x0>;
-	clocks = <&cps_syscon0 0  3>;
+	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
 };
 
@@ -91,7 +91,7 @@ spi@700680 {
 	#address-cells = <0x1>;
 	#size-cells = <0x0>;
 	cell-index = <0x0>;
-	clocks = <&cps_syscon0 0  3>;
+	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
 };
 
@@ -102,7 +102,7 @@ i2c@701000 {
 	#size-cells = <0>;
 	interrupts = <GIC_SPI 310 IRQ_TYPE_LEVEL_HIGH>;
 	timeout-ms = <1000>;
-	clocks = <&cps_syscon0 0  3>;
+	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
 };
 
@@ -113,7 +113,7 @@ i2c@701100 {
 	#size-cells = <0>;
 	interrupts = <GIC_SPI 311 IRQ_TYPE_LEVEL_HIGH>;
 	timeout-ms = <1000>;
-	clocks = <&cps_syscon0 0  3>;
+	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 0ad4169..d3202cc 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -151,7 +151,7 @@ spi@700600 {
 	#address-cells = <0x1>;
 	#size-cells = <0x0>;
 	cell-index = <0x0>;
-	clocks = <&cpm_syscon0 0  3>;
+	clocks = <&cpm_syscon0 1 21>;
 	status = "disabled";
 };
 
-- 
1.7.9.5

